projects/schematics/src/migrations/2_0/constructor-deprecations/data/star-rating.component.migration.ts

```diff
@@ -0,0 +1,20 @@
+import {
+  SPARTACUS_STOREFRONTLIB,
+  STAR_RATING_COMPONENT,
+  ANGULAR_CORE,
+  RENDERER_2,
+} from '../../../../shared/constants';
+import { ConstructorDeprecation } from '../../../../shared/utils/file-utils';
+
+export const STAR_RATING_COMPONENT_MIGRATION: ConstructorDeprecation = {
+  // projects\storefrontlib\src\shared\components\star-rating\star-rating.component.ts
+  class: STAR_RATING_COMPONENT,
+  importPath: SPARTACUS_STOREFRONTLIB,
+  deprecatedParams: [],
```

Comment:
In `deprecatedParams` you have to describe shape of the constructor you want to modify. Kinda misleading name of this property ;)
In this case add object with `ElementRef` and their import path.